.github/workflows/qat.yml

```diff
@@ -16,74 +16,64 @@ on:
-            echo "RUN_QAT=true" >> $GITHUB_ENV
-          else
-            echo "Not a feature branch or qat script not found. Skipping Playwright tests."
-            echo "RUN_QAT=false" >> $GITHUB_ENV
-          fi
-
-      - name: Check PMCQA
-        if: ${{ env.RUN_QAT == 'true' && inputs.THEME_NAME  != '' && inputs.QA_SITE_DOMAIN != '' }}
-        timeout-minutes: 10
-        run: |
-          # `version-hash` is created by jenkins script when a branch is pushed
-          # https://github.com/penske-media-corp/pmc-jenkins-scripts/blob/8ee0244a8485b61cfa6d3f9af00f7e6a55ddb042/bin/push-pmcqa-repo.sh#L36-L38
-          hash_url="https://${{ env.FEATURE_BRANCH_NAME }}.${{ inputs.QA_SITE_DOMAIN }}/wp-content-vipgo-sites/${{ env.FEATURE_BRANCH_NAME }}/themes/vip/${{ inputs.THEME_NAME }}/version-hash"
-          echo "HASH_URL: $hash_url"
-          hash=`curl --no-progress-meter --insecure $hash_url`
-          until [[ $hash == ${{ github.event.pull_request.head.sha }} ]]; do
-              echo -n "Waiting for Jenkins Deployment ... \n"
-              hash=`curl --no-progress-meter --insecure $hash_url`
-              sleep 5
-          done
-
-      - name: Install Playwright
-        if: ${{ env.RUN_QAT == 'true' }}
-        run: |
-          . "$NVM_DIR/nvm.sh"
-
-          nvm install
-
-          # Playwright needs to be added in package.json as dependency
-          npm ci
-
-      - name: Run Playwright tests
-        if: ${{ env.RUN_QAT == 'true' }}
-        run: |
-          if [ "${{ inputs.QA_SITE_DOMAIN }}" != '' ]; then
-            export ENVIRONMENT_URL="https://${{ env.FEATURE_BRANCH_NAME }}.${{ inputs.QA_SITE_DOMAIN }}"
-          fi
-          npm run ${{ env.NPM_QAT_SCRIPT }}
+          fetch-depth: 0 # Check out the full git history
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16 # Checkly CLI requires Node.js 16 or higher.
+      - name: Restore or cache node_modules # Restore node_modules cache if available
+        id: cache-node-modules
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: node-modules-${{ hashFiles('package-lock.json') }}
+      - name: Install dependencies # Install NPM dependencies if cache miss
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        run: npm ci
+      - name: Run checks # Run Checkly checks and record test session
+        id: run-checks
+        run: npx checkly test -e ENVIRONMENT_URL=${{ env.ENVIRONMENT_URL }} -e CHECKLY_REPO_SHA=${{ env.CHECKLY_REPO_SHA }} --reporter=github --record
+      - name: Create summary # Export Checkly report to GitHub job summary
+        if: always() # Always generate report even if tests fail
+        id: create-summary
+        run: cat checkly-github-report.md > $GITHUB_STEP_SUMMARY
+      - name: Deploy checks # Deploy checks to Checkly for debugging
+        if: always() # Deploy regardless of test result
```

Comment:
What if a new test being added is failing?

Reply:
The failing test will still fail but not block check from deploying to the Checkly account. That way, there is a test in the Checkly account that they can go look at for screenshots and debugging.

Reply:
I will test out both approaches and see if we can still get enough information to debug without the test deploying.

Reply:
Actually... we should only deploy on merges into main.

if a failing test blocks a PR, theoretically no failing tests would enter main.